publication-commons/src/main/java/no/unit/nva/publication/model/business/FilesApprovalThesis.java

```diff
@@ -0,0 +1,312 @@
+import static nva.commons.core.attempt.Try.attempt;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.Username;
+import no.unit.nva.model.associatedartifacts.file.File;
+import no.unit.nva.model.associatedartifacts.file.PendingFile;
+import no.unit.nva.publication.model.storage.FileApprovalThesisDao;
+import no.unit.nva.publication.model.storage.TicketDao;
+import no.unit.nva.publication.service.impl.ResourceService;
+import nva.commons.core.JacocoGenerated;
+
+@SuppressWarnings({"PMD.GodClass"})
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonTypeName(FilesApprovalThesis.TYPE)
+public class FilesApprovalThesis extends TicketEntry {
+
+    public static final String TYPE = "FilesApprovalThesis";
+    @JsonProperty(IDENTIFIER_FIELD)
+    private SortableIdentifier identifier;
+    @JsonProperty(STATUS_FIELD)
+    private TicketStatus status;
+    @JsonProperty(CUSTOMER_ID_FIELD)
+    private URI customerId;
+    @JsonProperty(MODIFIED_DATE_FIELD)
+    private Instant modifiedDate;
+    @JsonProperty(CREATED_DATE_FIELD)
+    private Instant createdDate;
+    @JsonProperty(WORKFLOW)
+    private PublishingWorkflow workflow;
+    @JsonProperty(ASSIGNEE_FIELD)
+    private Username assignee;
+    @JsonProperty(OWNER_AFFILIATION_FIELD)
+    private URI ownerAffiliation;
+    @JsonProperty(APPROVED_FILES_FIELD)
+    private Set<File> approvedFiles;
+    @JsonProperty(FILES_FOR_APPROVAL_FIELD)
+    private Set<File> filesForApproval;
+
+    public FilesApprovalThesis() {
+        super();
+    }
+
+    public static FilesApprovalThesis create(Resource resource, UserInstance userInstance,
+                                             PublishingWorkflow workflow) {
+        var fileApproval = createFileApproval(resource, userInstance, workflow);
+        return REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(workflow) ? completeAndApproveFiles(resource,
+                                                                                                   userInstance,
+                                                                                                   fileApproval)
+                   : handleMetadataOnlyWorkflow(resource, userInstance, workflow, fileApproval);
+    }
+
+    public void validateCreationRequirements(Resource resource) {
+        if (!resource.isDegree()) {
+            throw new IllegalStateException("FilesApprovalThesis requires publication to be a degree.");
+        }
+    }
+
+    @Override
+    public void validateCreationRequirements(Publication publication) {
```

Comment:
Can you add a note to the effect that this is left intentionally blank. You're encoding something from the business logic here that needs to be stated explicitly i.e. `// There are no validation requirements for creation of this ticket because anyone can do this, or it is done by a machine`

Reply:
I don't suggest this text, but whatever makes it so we don't have a body here. I guess also this code is actually called because of polymorphism?

Reply:
Maybe even make it Javadoc so the API exposes the fact that it is pointless to call this method and expect anything other than "meh".

Reply:
Added // NO OP as we did for all other tickets. I will try to extract common logic between PublishingRequest and this class to base class, so we do not duplicate code.